docs/shared/spm-setup-codegen-panel.mdx

```diff
@@ -0,0 +1,86 @@
+<ExpansionPanel title="Swift Package Manager">
+
+Apollo iOS exposes the Codegen CLI as an SPM plugin. This means you can run the Codegen CLI using the `swift package` command.
+
+> **Note:** SPM Plugins require permission to write files to your project directory. Each time you run these commands, you will be prompted to give the plugin permission. To bypass this prompt, we reccomend including the `--allow-writing-to-package-directory` flag before the plugin commmand name.
+
+<ExpansionPanelList>
+<ExpansionPanelListItem>
+
+#### Initialize the codegen configuration
+
+The Codegen CLI uses a JSON file to configure the code generation engine. Initializing the Codegen CLI creates this file with default values. From the directory of your `Package.swift` file, run:
+
+```bash
+swift package --allow-writing-to-package-directory apollo-initialize-codegen-config --schema-name ${MySchemaName}
+```
+
+`${MySchemaName}` should be the name you would like to be used as the namespace for your generated schema files.
+
+This will create an `apollo-codegen-config.json` file with default options.
+
+</ExpansionPanelListItem>
+
+<ExpansionPanelListItem>
+
+#### Set codegen configuration options
+
+Open the `apollo-codegen-config.json` file to configure code generation for your project.
+
+The default configuration will:
+- Find all GraphQL schema files ending with the file extension `.graphqls` within your project directory.
```

Comment:
is this a typo? should it be `.graphql`?

Reply:
Nope, SDL (schema definition) files generally have the `.graphqls` (s standing for schema) file extension!